packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps, DOMRef, DOMRefValue, FocusEvents, FocusableRef, InputDOMProps, ValueBase} from '@react-types/shared';
+import {centerBaseline} from './CenterBaseline';
+import {ContextValue, Radio, RadioGroup, RadioProps, Provider, RadioGroupStateContext, useSlottedContext} from "react-aria-components"
+import {createContext, forwardRef, ReactNode, useContext, useRef, useCallback, useEffect} from 'react';
+import {focusRing, StyleProps, getAllowedOverrides} from './style-utils' with {type: 'macro'};
+import {IconContext} from './Icon';
+import {style} from '../style/spectrum-theme' with {type: 'macro'};
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SegmentedControlProps extends ValueBase<string|null, string>, InputDOMProps, FocusEvents, StyleProps, AriaLabelingProps {
```

Comment:
I think we'll need `size` also

Reply:
so we removed the non-track variant since it wasn't accessible. originally it defined a prop like `trackStyle: 'none' | 'track'` but we've removed that in favor of just defaulting to the track style for now until design figures out the accessibility. 

and right now i don't see any t-shirt sizing in the figma designs. it's still under review so they might add it later